lib/spree/chimpy/interface/orders.rb

```diff
@@ -1,102 +1,257 @@
 module Spree::Chimpy
   module Interface
     class Orders
+      include Rails.application.routes.url_helpers
       delegate :log, to: Spree::Chimpy
 
       def initialize
-        @api = Spree::Chimpy.api
       end
 
       def api_call
-        @api.ecomm
+        Spree::Chimpy.api.ecommerce.stores(Spree::Chimpy::Config.store_id)
       end
 
       def add(order)
-        if source = order.source
-          info = Spree::Chimpy.list.info(source.email_id)
-          # use the one from mail chimp or fall back to the order's email
-          # happens when this is a new user
-          expected_email = (info[:email] || order.email).to_s
-        else
-          expected_email = order.email
+        ensure_products(order)
```

Comment:
Since we are passing the order around quite a bit here, does it make sense to create an iVar here and an attr_reader to make it available throughout the class without passing?